modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/JavascriptClientCodegen.java

```diff
@@ -289,14 +298,35 @@ public String escapeReservedWord(String name) {
         return "_" + name;
     }
 
+    private String createPath(String... segments) {
```

Comment:
createPath(String...) concatenates an array of path segments to return a path in which segments are joined by the correct platform-specific separator character. It also ignores empty and '.' segments. The new method is used in several places to replace code that used to yield paths with (potentially) a mixture of separator characters.

I will add a Javadoc comment now - **please hold off the merge** for now and I will update the PR accordingly.